File: src/main/java/com/example/application/views/LoginView.java
package com.example.application.views;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.login.AbstractLogin;
import com.vaadin.flow.component.login.LoginForm;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.PageTitle;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.server.VaadinSession;

@Tag("sa-login-view")
@Route(value = LoginView.ROUTE)
@PageTitle("Login")
public class LoginView extends VerticalLayout {
    public static final String ROUTE = "login";

    public LoginView() {
        super();

        LoginForm form = new LoginForm();
        form.addLoginListener(e -> {
            boolean isAuthenticated = authenticate(e);
            if (isAuthenticated) {
                form.getUI().ifPresent(ui -> ui.navigate(""));
            } else {
                form.setError(true);
            }
        });

        add(form);
    }

    private boolean authenticate(AbstractLogin.LoginEvent e) {
        VaadinSession session = getUI().get().getSession();
        return true;
    }


}

File: src/main/kotlin/com/example/application/views/LoginView.kt
package com.example.application.views

import com.example.application.backend.AuthService
import com.vaadin.flow.component.Tag
import com.vaadin.flow.component.login.AbstractLogin
import com.vaadin.flow.component.login.LoginForm
import com.vaadin.flow.component.orderedlayout.VerticalLayout
import com.vaadin.flow.router.PageTitle
import com.vaadin.flow.router.Route

@Tag("sa-login-view")
@Route(value = "login")
@PageTitle("Login")
class LoginView : VerticalLayout() {
    init {

        val form = LoginForm()
        form.addLoginListener { e ->
            val isAuthenticated = authenticate(e)
            if (isAuthenticated) {
                form.ui.ifPresent { ui -> ui.navigate("") }
            } else {
                form.isError = true
            }
        }

        add(form)
    }

    private fun authenticate(e: AbstractLogin.LoginEvent): Boolean {
        val username = e.username
        val password = e.password

        if (AuthService.authenticate(username, password)) {
            val session = ui.get().session
            session.setAttribute("user", username)
            session.setAttribute("password", password)
            return true
        }

        return false
    }
}
